src/main/kotlin/com/manimdsl/frontend/Type.kt

```diff
@@ -52,7 +59,16 @@ data class StackType(
     override fun getMethodByName(method: String): DataStructureMethod {
         return methods.getOrDefault(method, ErrorMethod)
     }
+
+    override fun toString(): String {
```

Comment:
Can this be removed?

Reply:
I put the `toString` there to make the semantic error message more readable but I can remove all of them if we don't want them

Reply:
There's no issue with making it more readable - it's just that this is the generic `DataStructureMethod` interface, representing `pop`, `push` etc, as well as any data structure methods we decide to add in the future. Where is this used in the semantic error handler?

Reply:
Oh sorry I thought this was only for the stack type so when we have incompatible type, the error message would just say "stack type" instead of the actual `toString` that prints the method and everything too. I will double check if I added this in the wrong place.

Reply:
It's currently only used for stack methods but we will probably end up being used for all methods on our language data structures

Reply:
I'm not exactly sure why the `toString` would cause a problem I thought it only overrides the `toString` for the type not the methods but if this is going to cause problems, I can remove all the `toString` I added.

Reply:
Oh no my bad - I thought it was inside `DataStructureMethod` because of the GitHub view 😅  ignore me, that's absolutely fine